Assets/Scripts/Models/RoomManager.cs

```diff
@@ -0,0 +1,328 @@
+﻿#region License
+// ====================================================
+// Project Porcupine Copyright(C) 2016 Team Porcupine
+// This program comes with ABSOLUTELY NO WARRANTY; This is free software, 
+// and you are welcome to redistribute it under certain conditions; See 
+// file LICENSE, which is part of this source code package, for details.
+// ====================================================
+#endregion
+using System;
+using System.Collections.Generic;
+
+public class RoomManager
+{
+    public event Action<Room> Adding;
+
+    public event Action<Room> Added;
+
+    public event Action<Room, Room> Split;
+
+    public event Action<Room, Room> Joined;
+
+    public event Action<Room> Removing;
+
+    public event Action<Room> Removed;
+
+    public List<Room> rooms;
+
+    public RoomManager()
+    {
+        rooms = new List<Room>();
+    }
+
+    public Room GetOutsideRoom()
+    {
+        return rooms[0];
+    }
+
+    public int GetRoomID(Room r)
```

Comment:
No single letter variables. Yes...yes... there are a lot of them in code. But if you change code - make it even better.